src/FffGcodeWriter.h

```diff
@@ -483,13 +483,23 @@ class FffGcodeWriter : public SettingsMessenger, NoCopy
     bool processSkinPart(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part, unsigned int layer_nr, int skin_overlap, int infill_angle) const;
 
     /*!
-     * Add the gcode of the top/bottom skin of the given skin part and of the perimeter gaps.
+     * Add the extra skin walls
      * 
-     * Perimeter gaps are handled for skin outlines and printed after the skin fill of the skin part is printed.
+     * \param[in] storage where the slice data is stored.
+     * \param gcode_layer The initial planning of the gcode of the layer.
+     * \param mesh The mesh for which to add to the layer plan \p gcode_layer.
+     * \param extruder_nr The extruder for which to print all features of the mesh which should be printed with this extruder
+     * \param mesh_config the line config with which to print a print feature
+     * \param skin_part The skin part for which to create gcode
+     * \param layer_nr The current layer number.
+     * \param[out] added_something Whether this function added anything to the layer plan
+     */
+    void processSkinInsets(const SliceDataStorage& storage, LayerPlan& gcode_layer, const SliceMeshStorage& mesh, const int extruder_nr, const PathConfigStorage::MeshPathConfigs& mesh_config, const SkinPart& skin_part, unsigned int layer_nr, bool& added_something) const;
+
+    /*!
+     * Add the normal skinfill which is the area inside the innermost skin inset
      * 
-     * Note that the normal perimeter gaps are printed with the outer wall extruder,
-     * while newly generated perimeter gaps between consecutive insets of a concentric top/bottom pattern
-     * are printed with the top bottom extruder.
+     * Perimeter gaps are generated when the pattern is concentric
```

Comment:
No. The Infill.generate processes areas into lines/polygons and generates new areas for the gaps created by the infill pattern when it's concentric. These gaps aren't processed here. They are generated here. They are processed in `processSkinPart`.

This will make a bit more sense in CURA-3940, because then we add the topmost_skin which also generated gaps which are handled in processSkinPart.